File: tests/adapters/rust/goldens/literals/max_tokens_10.rs
// Rust module for testing literal optimization.

use std::collections::HashMap;

// Short string literal (should be preserved)
const SHORT_MESSAGE: &str = "Hello, World!";

// Long string literal (candidate for trimming)
const LONG_MESSAGE: &str = "This is an extremely long message that contains a…"; // literal string (−62 tokens)

// Multi-line string with formatting
const TEMPLATE_WITH_DATA: &str = r#"User Information:
- Nam…"#; // literal string (−33 tokens)

#[derive(Debug)]
struct DataContainer {
    // Small vec (should be preserved)
    tags: Vec<String>,

    // Large vec (candidate for trimming)
    items: Vec<String>,

    // Small map (should be preserved)
    metadata: HashMap<String, String>,

    // Large map (candidate for trimming)
    configuration: HashMap<String, HashMap<String, i32>>,
}

struct LiteralDataManager {
    // Small config (should be preserved)
    small_config: HashMap<String, bool>,

    // Large config (candidate for trimming)
    large_config: HashMap<String, HashMap<String, serde_json::Value>>,

    // Supported languages
    supported_languages: Vec<String>,

    // Allowed extensions
    allowed_extensions: Vec<String>,
}

impl LiteralDataManager {
    fn new() -> Self {
        let mut small_config = HashMap::new();
        small_config.insert("debug".to_string(), true);
        small_config.insert("version".to_string(), false);

        let large_config = {
            let mut config = HashMap::new();
            // …
            config
        }; // literal array (−444 tokens)

        let supported_languages = vec![
            "english",
            "spanish",
            "…",
        ].into_iter().map(String::from).collect(); // literal array (−89 tokens)

        let allowed_extensions = vec![
            ".rs",
            ".py",
            "…",
        ].into_iter().map(String::from).collect(); // literal array (−62 tokens)

        Self {
            small_config,
            large_config,
            supported_languages,
            allowed_extensions,
        }
    }

    fn process_data(&self) -> DataContainer {
        let small_vec = vec!["one".to_string(), "…"]; // literal array (−9 tokens)

        let large_vec = vec![
            "item_001",
            "…",
        ].into_iter().map(String::from).collect(); // literal array (−146 tokens)

        let mut metadata = HashMap::new();
        metadata.insert("type".to_string(), "test".to_string());
        metadata.insert("count".to_string(), "3".to_string());

        DataContainer {
            tags: small_vec,
            items: large_vec,
            metadata,
            configuration: HashMap::new(),
        }
    }

    fn get_long_query(&self) -> &str {
        r#"
SELECT
    users.id, users.u…"# // literal string (−168 tokens)
    }
}

// Module-level constants with different sizes
struct SmallConstants;
impl SmallConstants {
    const API_VERSION: &'static str = "v1";
    const DEFAULT_LIMIT: i32 = 50;
}

lazy_static! {
    static ref HTTP_STATUS_CODES: HashMap<&'static str, i32> = {
        let mut m = HashMap::new();
        // …
        m
    };
} // literal object (−196 tokens)

lazy_static! {
    static ref ERROR_MESSAGES: HashMap<&'static str, &'static str> = {
        let mut m = HashMap::new();
        // …
        m
    };
} // literal object (−113 tokens)

fn main() {
    let manager = LiteralDataManager::new();
    let data = manager.process_data();

    println!("Tags: {:?}", data.tags);
    println!("Items count: {}", data.items.len());
    println!("{}", manager.get_long_query());
}

File: tests/adapters/go/goldens/literals/max_tokens_10.go
// Go module for testing literal optimization.
package main

import (
	"fmt"
)

// Short string literal (should be preserved)
const ShortMessage = "Hello, World!"

// Long string literal (candidate for trimming)
const LongMessage = "This is an extremely long message that contains a…" // literal string (−63 tokens)

// Multi-line string with formatting
const TemplateWithData = `User Inform…` // literal string (−50 tokens)

// DataContainer holds various literal types
type DataContainer struct {
	// Small slice (should be preserved)
	Tags []string

	// Large slice (candidate for trimming)
	Items []string

	// Small map (should be preserved)
	Metadata map[string]interface{}

	// Large map (candidate for trimming)
	Configuration map[string]interface{}
}

// LiteralDataManager manages literal data
type LiteralDataManager struct {
	// Small config (should be preserved)
	smallConfig map[string]interface{}

	// Large config (candidate for trimming)
	largeConfig map[string]interface{}

	// Supported languages
	supportedLanguages []string

	// Allowed extensions
	allowedExtensions []string
}

// NewLiteralDataManager creates a new manager instance
func NewLiteralDataManager() *LiteralDataManager {
	return &LiteralDataManager{
		smallConfig: map[string]interface{}{
		    "debug":   true,
		    // … (1 more, −10 tokens)
		},
		largeConfig: map[string]interface{}{
		    "database": map[string]interface{}{
		        "host": "localhost",
		        // … (5 more, −107 tokens)
		    },
		    // … (3 more, −334 tokens)
		},
		supportedLanguages: []string{
		    "english",
		    // … (23 more, −89 tokens)
		},
		allowedExtensions: []string{
		    ".go",
		    // … (20 more, −62 tokens)
		},
	} // literal struct (−495 tokens)
}

// ProcessData creates a data container with various literals
func (m *LiteralDataManager) ProcessData() *DataContainer {
	smallSlice := []string{"one", "two", "three"}

	largeSlice := []string{
		"item_001",
		"…"
	} // literal slice (−145 tokens)

	nestedData := map[string]interface{}{
		"level1": map[string]interface{}{
		    "level2": map[string]interface{}{
		        "level3": map[string]interface{}{
		            "data": []map[string]interface{}{
		                {"id": 1, "name": "First", "active": true},
		                // … (4 more, −69 tokens)
		            },
		            // … (1 more, −126 tokens)
		        },
		    },
		},
	}

	return &DataContainer{
		Tags:          smallSlice,
		Items:         largeSlice,
		Metadata:      map[string]interface{}{"type": "test", "count": len(smallSlice)},
		Configuration: nestedData,
	}
}

// GetLongQuery returns a very long SQL-like query string
func (m *LiteralDataManager) GetLongQuery() string {
	return `
		SELECT…` // literal string (−191 tokens)
}

// Module-level constants with different sizes
var SmallConstants = struct {
	APIVersion   string
	DefaultLimit int
}{
	APIVersion:   "v1",
	DefaultLimit: 50,
}

var HTTPStatusCodes = map[string]int{
	"CONTINUE":                      100,
	// … (40 more, −317 tokens)
}

var ErrorMessages = map[string]string{
	"VALIDATION_FAILED":      "Input validation failed. Please check you…" // literal string (−6 tokens),
	// … (6 more, −102 tokens)
}

func main() {
	manager := NewLiteralDataManager()
	data := manager.ProcessData()

	fmt.Printf("Tags: %v\n", data.Tags)
	fmt.Printf("Items count: %d\n", len(data.Items))
	fmt.Println(manager.GetLongQuery())
}
